[dx-732] release note 5.2.1, no docker link (#3306)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.2.md

---
title: Tyk Gateway 5.2 Release Notes
date: 2023-09-27T15:49:11Z
description: "Release notes for Tyk Gateway version 5.2"
tags: ["Release notes", "v5.2", "5.2.0", "5.2", "changelog"]
---

**Open Source**

##### This page contains all release notes for version 5.2.X displayed in a reverse chronological order

### Support Lifetime

> Our minor releases are supported until our next minor comes out. This would be R5.3 scheduled in Q4 if this goes ahead as planned. If not, 5.2 will remain in support until our next LTS version comes out in March 2024.

Reviewer: As for the Dashboard, no R5.3 is currently scheduled.

Author: @dcs3spp

Reviewer 2: Updated support lifetime statement for Tyk Gateway

### Upgrading Tyk
Please refer to the [upgrading Tyk]({{< ref "upgrading-tyk" >}}) page for further guidance with respect to the upgrade strategy.

---

## 5.2.1 Release Notes 

> ##### Release Date 29 Sep 2023

Reviewer: There's no "Release Date" on the Dashboard page - is this deliberate?

It's also the wrong date :)

Author: fixed already

### Breaking Changes
This release has no breaking changes.

### Deprecations
There are no deprecations in this release.

### Release Highlights
This release primarily focuses on bug fixes.
For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.2.0">}}) below.

### Downloads
- [docker image to pull](https://hub.docker.com/layers/tykio/tyk-gateway/v5.2.1/images/blabla)
- [source code](https://github.com/TykTechnologies/tyk/releases/tag/v5.1.2)

### Changelog {#Changelog-v5.2.1}

#### Changed (3 change):

> - Enhance message quality by eliminating unnecessary messages

Reviewer: I feel this needs more context - which messages? where?

> - Fixed a bug that occurs during Gateway reload where the Gateway would continue to load new API definitions even if policies failed to load; this would leave the customer at risk of an APIs being invoked without the associated policies. Tyk Now Tyk offers configurable retries for resource loading, ensuring a specified number of attempts to load resources (APIs and policies). If a resource fails to load, an error will be logged and the Gateway reverts to its last working configuration.

Reviewer: ```suggestion
- Fixed a bug that occurs during Gateway reload where the Gateway would continue to load new API definitions even if policies failed to load. This led to a risk that an API could be invoked without the associated policies (for example, describing access control or rate limts) having been loaded. Now Tyk offers a configurable retry for resource loading, ensuring that a specified number of attempts will be made to load resources (APIs and policies). If a resource fails to load, an error will be logged and the Gateway reverts to its last working configuration.
```

We have introduced two new variables to configure this behaviour:
  - `resource_sync.retry_attempts` - defines the number of retries that the Gateway should perform during a resource sync (APIs or policies), defaulting to zero which means no retries are attempted
  - `resource_sync.interval` - setting the fixed interval between retry attempts (in seconds)"

> - We've included the previously missing attributes, `http.response.body.size` and `http.request.body.size`, in both Tyk HTTP spans and upstream HTTP spans. This addition enables users to gain better insight into request and response sizes within their traces.

Reviewer: I feel this needs some introduction to explain that it relates to OTel (does it?)

Author: check my change below

> - Fixed a bug where *Tyk Dashboard* or *Tyk Gateway* APIs would not allow you to modify a key that provides access rights to an inactive API

Reviewer: ```suggestion
- Fixed a bug where, if you created a Key which provided access to an inactive or draft API, you would be unable to subsequently modify that Key (via the Dashboard or directly via the Tyk Gateway API)
```

I think this should be in "Fixed" not "Changed"

Author: why "via dashboard" without the word "api"? did you want it to include ui and api?

Author: also `Key` needs to be lowercase

Reviewer: I was trying to show that it can be via Tyk Dashboard API or Tyk Dashboard UI... but you're right, we should be explicit.

```suggestion
- Fixed a bug where, if you created a key which provided access to an inactive or draft API, you would be unable to subsequently modify that Key (via the Tyk Dashboard UI, Tyk Dashboard API or Tyk Gateway API)
```

#### Fixed (6 changes):
- Fixed a memory leak that occurred when enabling the [strict routes option]({{< ref "tyk-oss-gateway/configuration#http_server_optionsenable_strict_routes" >}}) to change the routing to avoid nearest-neighbour requests on overlapping routes (`TYK_GW_HTTPSERVEROPTIONS_ENABLESTRICTROUTES`)

- Fixed a potential performance issue related to high rates of *Tyk Gateway* reloads (when the Gateway is updated due to a change in APIs and/or policies). The gateway uses a timer that ensures there's at least one second between reloads, however in some scenarios this could lead to poor performance (for example overloading Redis). We have introduced a new configuration option `reload_interval` (`TYK_GW_RELOADINTERVAL`) that can be used to adjust the duration between reloads and hence optimise the performance of your Tyk deployment.

> - Fixed an issue in GraphQL APIs, where [headers]({{< ref "graphql/gql-headers" >}}) were not properly forwarded upstream for [GQL/UDG subscriptions]({{< ref "getting-started/key-concepts/graphql-subscriptions" >}}).

Reviewer: ```suggestion
- Fixed an issue with GraphQL APIs, where [headers]({{< ref "graphql/gql-headers" >}}) were not properly forwarded upstream for [GQL/UDG subscriptions]({{< ref "getting-started/key-concepts/graphql-subscriptions" >}}).
```

- Fixed a bug where the Gateway did not correctly close idle upstream connections (sockets) when configured to generate a new connection after a configurable period of time (using the [max_conn_time]({{<ref "tyk-oss-gateway/configuration#max_conn_time" >}})
configuration option). This could lead to the Gateway eventually running out of sockets under heavy load, impacting performance.

> - Remove the extra chunked transfer encoding that was added unnecessarily to `rawResponse` analytics

Reviewer: ```suggestion
- Removed the extra chunked transfer encoding that was added unnecessarily to `rawResponse` analytics
```

> - Resolved a bug in HTTP GraphQL APIs where, when the [Persist GraphQL middleware]({{< ref "graphql/persisted-queries" >}}) was used in combination with [Response Body Transform]({{< ref "advanced-configuration/transform-traffic/response-body" >}}), the response's body transformation was not being executed.

Reviewer: ```suggestion
- Resolved a bug with HTTP GraphQL APIs where, when the [Persist GraphQL middleware]({{< ref "graphql/persisted-queries" >}}) was used in combination with [Response Body Transform]({{< ref "advanced-configuration/transform-traffic/response-body" >}}), the response's body transformation was not being executed.
```

{{< img src="img/bugs/bug-persistent-gql.png" width="400" alt="Bug in persistent gql and response body transform" title="The setup of graphQL middlewares">}}

> #### Dependencies (1 change):

Author: @dcs3spp can you please remind me why you want to remove it

> #### Dependencies (1 change):

Reviewer 2: Suggested to remove since the changelog  is under the heading of the Patch release note title

> - Update TykTechnologies/gorm to v1.21 in Tyk Gateway 

Reviewer: ```suggestion
- Updated TykTechnologies/gorm to v1.21 in Tyk Gateway 
```

We have introduced two new variables to configure this behaviour:
  - `resource_sync.retry_attempts` - defines the number of retries that the Gateway should perform during a resource sync (APIs or policies), defaulting to zero which means no retries are attempted
  - `resource_sync.interval` - setting the fixed interval between retry attempts (in seconds)"

- We've included the previously missing attributes, `http.response.body.size` and `http.request.body.size`, in both Tyk HTTP spans and upstream HTTP spans. This addition enables users to gain better insight into request and response sizes within their traces.

- Fixed a bug where *Tyk Dashboard* or *Tyk Gateway* APIs would not allow you to modify a key that provides access rights to an inactive API

> ##### Fixed (6 changes):

Reviewer 3: ```suggestion
##### Fixed (7 changes):

- Fixed a memory leak issue in Gateway 5.2.0 if opentelemetry is enabled. It was caused by multiple otelhttp handler being created. We have updated to use single instance of otelhttp handler in 5.2.1 to improve performance under high traffic load.

```

description: "Release notes documenting updates, enhancements, and changes for Tyk Gateway versions within the 5.2.X series."
tags: ["Tyk Gateway", "Release notes", "v5.2", "5.2.0", "5.2", "changelog", "5.2.1"]
---

**Open Source** ([Mozilla Public License](https://github.com/TykTechnologies/tyk/blob/master/LICENSE.md))

**This page contains all release notes for version 5.2.X displayed in reverse chronological order**

### Support Lifetime
Minor releases are supported until our next minor comes out. There is no 5.3 scheduled in Q4. Subsequently, 5.2 will remain in support until our next LTS version comes out in March 2024.

---

## 5.2.1 Release Notes 

##### Release Date 10 Oct 2023 <<update>>

#### Breaking Changes
This release has no breaking changes.

#### Deprecations
There are no deprecations in this release.

#### Upgrade instructions
If you are on a 5.2.0 we advise you to upgrade ASAP and if you are on an older version skip 5.2.0 and upgrade directly to this release.

#### Release Highlights
This release primarily focuses on bug fixes.
For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.2.0">}}) below.

#### Downloads

> <<- [docker image to pull](https://hub.docker.com/layers/tykio/tyk-gateway/v5.2.1/images/blabla)>>

Reviewer 2: ```suggestion
- [docker image to pull](https://hub.docker.com/layers/tykio/tyk-gateway/v5.2.1/images/sha256-47cfffda64ba492f79e8cad013a476f198011f5a97cef32464f1f47e1a9be9a2?context=explore)
```

- [source code](https://github.com/TykTechnologies/tyk/releases/tag/v5.1.2)

#### Changelog {#Changelog-v5.2.1}

##### Changed (4 change):
- Enhance log message quality by eliminating unnecessary messages

> - Fixed a bug that occurs during Gateway reload where the Gateway would continue to load new API definitions even if policies failed to load. This led to a risk that an API could be invoked without the associated policies (for example, describing access control or rate limts) having been loaded. Now Tyk offers a configurable retry for resource loading, ensuring that a specified number of attempts will be made to load resources (APIs and policies). If a resource fails to load, an error will be logged and the Gateway reverts to its last working configuration.

Author: ```suggestion
- Fixed a bug that occurs during Gateway reload where the Gateway would continue to load new API definitions even if policies failed to load. This led to a risk that an API could be invoked without the associated policies (for example, describing access control or rate limits) having been loaded. Now Tyk offers a configurable retry for resource loading, ensuring that a specified number of attempts will be made to load resources (APIs and policies). If a resource fails to load, an error will be logged and the Gateway reverts to its last working configuration.
```

We have introduced two new variables to configure this behaviour:
  - `resource_sync.retry_attempts` - defines the number of retries that the Gateway should perform during a resource sync (APIs or policies), defaulting to zero which means no retries are attempted
  - `resource_sync.interval` - setting the fixed interval between retry attempts (in seconds)"

> - We've included the previously missing attributes, `http.response.body.size` and `http.request.body.size`, in both Tyk HTTP spans and upstream HTTP spans. This addition enables users to gain better insight into request and response sizes within their traces.

Author: ```suggestion
- For OpenTelemetry users, we've included much-needed attributes, `http.response.body.size` and `http.request.body.size`, in both Tyk HTTP spans and upstream HTTP spans. This addition enables users to gain better insight into incoming/outgoing request/response sizes within their traces.
```

Author: @caroltyk @SonjaChevre what do you think about this? 
Also - have we updated the doc with this? (https://tyk.io/docs/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview/)
